cli-plugins/examples/helloworld/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	cliplugins "github.com/docker/cli/cli-plugins"
+	"github.com/docker/cli/cli-plugins/plugin"
```

Comment:
I didn't check the code and it's structure yet, but these imports (`docker/cli/cli-plugins` and especially `docker/cli/cli-plugins/plugin`) feel weird. Just having this first look, I think the Run function and the cliplugins.Metadata should be in the same package.

Reply:
IIRC that caused a circular include loop. I can certainly try and untangle it again though.

See also https://github.com/docker/cli/pull/1564#issuecomment-446963707 and the final bullet in the intro comment -- I'm awaiting MAINTAINERS guidance on the overall package naming.

Reply:
Moving metadata.go into `docker/cli/cli-plugins/plugin` in a quick and dirty way:
```
import cycle not allowed
package github.com/docker/cli/cmd/docker
	imports github.com/docker/cli/cli
	imports github.com/docker/cli/cli-plugins/manager
	imports github.com/docker/cli/cli-plugins/plugin
	imports github.com/docker/cli/cli
```

But moving it into `docker/cli/cli-plugins/manager` seems to have worked (at least the quick and dirty version), so I'll go with that for now (pending a renaming of the `cli-plugins` dir).

Reply:
Is naming it `cli-plugin` potentially dangerous;

IIRC, Go treats

```go
import foobar.com/foo/package-name
```` 

the same as

```go
import name foobar.com/foo/package-name
```

(stripping the `package-` prefix, making it an implicit alias)

Reply:
IIRC you get whatever the `package foo` in the go files says which is distinct from the path you used to get to import it (customarily they are the same, of have a prefix stripped).

NB in the latest code there is nothing directly in `cli-plugins`, it's all subpackages so the thing you worry about doesn't crop up.